src/libraries/RegistryHashes.sol

```diff
@@ -3,44 +3,304 @@ pragma solidity ^0.8.20;
 
 import {IDelegateRegistry} from "../IDelegateRegistry.sol";
 
+/**
+ * @title Library for calculating the hashes and storage locations used in the delegate registry
+ *
+ * The encoding for the 5 types of delegate registry hashes should be as follows
+ *
+ * ALL:         keccak256(abi.encode(from, rights, to))
+ * CONTRACT:    keccak256(abi.encode(from, rights, to, contract_))
+ * ERC721:      keccak256(abi.encode(from, rights, to, tokenId, contract_))
+ * ERC20:       keccak256(abi.encode(from, rights, to, contract_))
```

Comment:
Is there a potential collision here when somebody delegateForContract() and delegateForERC20() on the same contract?

Reply:
They do collide for the same inputs, but the hash is different since we overwrite the last byte to the type

Reply:
Ah of course